tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedEventMatcher.cs

```diff
@@ -0,0 +1,167 @@
+
+            for (int i = 0; i < actualEvents.Count; i++)
+            {
+                var actualEvent = actualEvents[i];
+                var expectedEventDocument = expectedEventsDocuments[i].AsBsonDocument;
+                if (expectedEventDocument.ElementCount != 1)
+                {
+                    throw new FormatException("Expected event document model must contain a single element");
+                }
+                var expectedEventType = expectedEventDocument.GetElement(0).Name;
+                var expectedEventValue = expectedEventDocument[0].AsBsonDocument;
+
+                switch (actualEvent)
+                {
+                    case CommandStartedEvent commandStartedEvent:
+                        expectedEventType.Should().Be("commandStartedEvent");
+                        foreach (var element in expectedEventValue)
+                        {
+                            switch (element.Name)
+                            {
+                                case "command":
+                                    _valueMatcher.AssertValuesMatch(commandStartedEvent.Command, element.Value);
+                                    break;
+                                case "commandName":
+                                    commandStartedEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                case "databaseName":
+                                    commandStartedEvent.DatabaseNamespace.DatabaseName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'");
+                            }
+                        }
+                        break;
+                    case CommandSucceededEvent commandSucceededEvent:
+                        expectedEventType.Should().Be("commandSucceededEvent");
+                        foreach (var element in expectedEventValue)
+                        {
+                            switch (element.Name)
+                            {
+                                case "reply":
+                                    _valueMatcher.AssertValuesMatch(commandSucceededEvent.Reply, element.Value);
+                                    break;
+                                case "commandName":
+                                    commandSucceededEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'");
+                            }
+                        }
+                        break;
+                    case CommandFailedEvent commandFailedEvent:
+                        expectedEventType.Should().Be("commandFailedEvent");
```

Comment:
Replace 105-106 with:

    case "commandFailedEvent":
        var commandFailedEvent = actualEvent.Should().BeOfType<CommandFailedEvent>().Subject;